client/checkout/platform-checkout/index.js

```diff
@@ -23,6 +23,8 @@ const renderSaveUserSection = () => {
 	)?.[ 0 ];
 	const buttonParent = placeOrderButton?.parentNode;
 	const checkoutPageSaveUserContainer = document.createElement( 'div' );
+	checkoutPageSaveUserContainer.className =
+		'platform-checkout-save-new-user-container';
```

Comment:
I see. Since we use the class `platform-checkout-save-new-user-container` to determine whether or not `saveUserSection` is already created (L14 above), by moving the classname to here, we make sure that the mount point is independent of the React component. 

Regardless of how many payment methods and how many times this is called, we will either have `<div class="'platform-checkout-save-new-user-container'"></div>` or with the component mounted inside.

This looks good to me.